Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -254,30 +281,50 @@ - (void)enumerateObjectsWithOptions:(NSEnumerationOptions)opts usingBlock:(void
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (id)objectAtIndex:(NSUInteger)index {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (index >= [self count]) {
+        [NSException raise:NSRangeException format:@"index is greater than the OrsdereSet size %d", [self count]];
+    }
+    return *std::next(setContainer.begin(), index);
```

Comment:
what're the advantages to doing this versus setContainer[index]? operator[] for a vector should be slightly lighter-weight than advancing the begin iterator